system-tests-new/svn_fcm/test_svn_fcm.py

```diff
@@ -0,0 +1,191 @@
+# ##############################################################################
+#  (c) Crown copyright Met Office. All rights reserved.
+#  For further details please refer to the file COPYRIGHT
+#  which you should have received as part of this distribution
+# ##############################################################################
+"""
+Test svn and fcm steps, if their underlying cli tools are available.
+
+"""
+import shutil
+from pathlib import Path
+from unittest import mock
+import warnings
+
+import pytest
+
+from fab.steps.grab.fcm import FcmCheckout, FcmExport, FcmMerge
+from fab.steps.grab.svn import GrabSvnBase, SvnCheckout, SvnExport, SvnMerge
+
+# Fcm isn't available in the github test images...unless we install it from github.
+
+# Which tools are available?
+export_classes = [i for i in [SvnExport, FcmExport] if i.tool_available()]
+checkout_classes = [i for i in [SvnCheckout, FcmCheckout] if i.tool_available()]
+merge_classes = [i for i in [SvnMerge, FcmMerge] if i.tool_available()]
+if not export_classes:
+    warnings.warn('Neither svn not fcm are available for testing')
+
+
+@pytest.fixture
+def repo_url(tmp_path):
+    shutil.unpack_archive(
+        Path(__file__).parent / 'repo.tar.gz',
+        tmp_path)
+    return f'file://{tmp_path}/repo'
+
+
+@pytest.fixture
+def config(tmp_path):
+    return mock.Mock(source_root=tmp_path / 'fab_proj/source')
+
+
+class TestFcmExport(object):
+
+    # Run the test twice, once with SvnExport and once with FcmExport - depending on which tools are available.
+    @pytest.mark.parametrize('export_class', export_classes)
+    def test_export(self, repo_url, config, export_class):
+        export = export_class(src=f'{repo_url}/proj/main/trunk', dst='proj')
+        export.run(artefact_store=None, config=config)
+
+        # Make sure we can export twice.
+        # Todo: should the export step wipe the destination first?
+        export.run(artefact_store=None, config=config)
+
+
+class TestFcmCheckout(object):
+
+    @pytest.mark.parametrize('checkout_class', checkout_classes)
+    def test_new_folder(self, repo_url, config, checkout_class):
+        checkout = checkout_class(src=f'{repo_url}/proj/main/trunk', dst='proj')
+        checkout.run(artefact_store=None, config=config)
+
+        file2_txt = open(config.source_root / 'proj/file2.txt').read()
+        assert "This is sentence two in file two." in file2_txt
```

Comment:
Might use `str.endswith()` to ensure the string is in the right place as well as present.